Fix CodeQL issues + private properties

Digitalone1 opened the pull request:
### Requirements 

* Filling out the template is required.
* All new code requires tests to ensure against regressions.
  - However, if your PR contains zero code changes, feel free to select the checkmark below to indicate so.

* [x] Have you ran tests against this code?

### Description of the Change

I wanted to contribute on this refactoring. Fixed CodeQL issues and introduced private properties.

It was a little tricky to add the private properties in the inherited constructor because `this` can be used only after the base constructor is executed, so I could not pass the private properties as argument.

That's because I wrongly intended it as an interface (or a trait), which is not, obviously. Anyway, to reuse the code, I thought it was better to make a method to initialize properties.

@confused-Techiet Tell me what do you think. If it's not worth, feel free to drop the last commits.

A question: @confused-Techie prefixing the `_` you meant the method as protected? Even if protected methods does not exist in JS.

confused-Techie reviewed (commented):
Thanks a ton for helping out on that refactor, I do want to add some pointers, with setting the public properties, that is a good idea and make things way cleaner. And depending on future usage could make our lives easier.

As for the getters and setters, those aren't actually needed for these values. And I think may make things behave a bit strange when checking types (iirc from past experience) since now the class has `this.apiUrl` and `this.apiUrl` as both a variable and setter. We should be fine leaving those as just variables.

As for your `_initializer` method I don't see this within the documentation for a class at all, is there something I'm missing?

The calling of `super()` in the extends class will automatically call the `constructor()` of it's parent class to where we don't need an extra function to assign anything, since any parameters provided to `super()` will be used in `constructor()` automatically.

Otherwise the other changes here look good, Like using the Public and private properties, and changing from `let` to `const` in some places. And thanks for fixing my snake case, seems to be a bad habit I have especially after adding it to our Guidelines.

Digitalone1 commented:
> As for the getters and setters, those aren't actually needed for these values. And I think may make things behave a bit strange when checking types (iirc from past experience) since now the class has `this.apiUrl` and `this.apiUrl` as both a variable and setter. We should be fine leaving those as just variables.

You are completely right. I misunderstood the behaviour of getters/setters. Besides I don't have to assign the value to itself in the setter because this lead to a loop calling the setter over and over. 

I will fix it. If you have the code by hand, you can remove them. 

> As for your `_initializer` method I don't see this within the documentation for a class at all, is there something I'm missing?

It's not specific to javascript, I added it as a method to i set/initialize the properties of the base class passing private properties of the derived class.   

> The calling of `super()` in the extends class will automatically call the `constructor()` of it's parent class to where we don't need an extra function to assign anything, since any parameters provided to `super()` will be used in `constructor()` automatically.

I know, but I wanted to call the base constructor passing the private properties (acting as default values) as argument to the derived class, but this is not allowed because `this` reference cannot be accessed before executing the base constructor. 

So I came up with the initializer method which get the private properties as default values for Github. 

If you think this is not needed, feel free to revert.

Digitalone1 commented:
So I removed the setters.

I'd like to do something like the following:
```js
super({
    api_url: opts?.api_url ?? this.#defaultApiUrl,
    ok_status: this.#defaultAcceptableStatusCodes
});
```

But it's not allowed because `this` in the derived is accessible only after the base constructor has been completely executed.

That's why I introduced the `_initializer` (which is not a special Javascript function, just a custom method). I intended it more like a [trait](https://www.php.net/manual/en/language.oop5.traits.php).

 I feel it's better to have privates:
```js
#defaultApiUrl = "https://api.github.com";
#defaultAcceptableStatusCodes = [200, 401];
```
at the start of each derived so we know clearly which are the defaults to be assigned to base properties rather than looking at the arguments of `super` inside the of constructor.

Again, if you think this is unneeded and overkill, you can revert.

confused-Techie reviewed (approved):
I won't lie, while initially may have looked a bit overkill, reading back through it now, now that it's been a little bit since I looked at the Git class itself, I totally see what you mean.

The readability is surprisingly hugely impacted by the small change of private variables. And while the `_initialized` may not be as clear of a convention for essentially calling a custom `super` that's what we need to do, so I feel it makes sense. And plus having the convention of psuedo-protected parent classes prefixed with `_` makes it easy enough.

Thanks a ton for the assist on this refactor, and looks great!
